File: python_backend/ranges/pawn/has_moved.py
from custom_except import *


def has_moved(pawn_hist):
    """takes a pawn history as input, returns true if that pawn has moved else returns false"""
    if len(pawn_hist) > 1:
        return True
    else:
        return False


if __name__ == "__main__":
    # test 1:
    print('test 1:')
    print(has_moved([(3, 2)]))

    # test 2:
    print('test 2:')
    print(has_moved([(3, 2), (3, 4)]))

File: app/src/game_logic/ranges/pawn/hasMoved.js


export function hasMoved(pawnHist) {
    /*takes a pawn history as input, returns true if that pawn has moved else returns false**/
    return pawnHist.length > 1;
}

// module.exports = hasMoved;

/** for node.js
if (require.main === module) {
    // test 1:
    console.log('test 1:')
    console.log(hasMoved([[3, 2]]))
    // test 2:
    console.log('test 2:')
    console.log(hasMoved([[3, 2], [3, 4]]))

}
*/